src/starknet-docker-proxy.ts

```diff
@@ -32,7 +32,17 @@ export class StarknetDockerProxy extends DockerServer {
             volumes.push("-v", `${mirroredPath}:${mirroredPath}`);
         }
 
-        return [...volumes, "--network", "host"];
+        const dockerArgs = [...volumes];
+        // Check host machine
+        const isArm64 = process.arch === "arm64";
+        if (isArm64) {
+            this.port = await getFreePort();
```

Comment:
As discussed, there should be only one place where `getFreePort` is called. We could introduce a `getPort` method which returns `this.port` if defined, otherwise calls `getFreePort` before that. The new method would have to return a Promise (i.e. be async).

Reply:
https://github.com/Shard-Labs/starknet-hardhat-plugin/pull/210/commits/73d4039c6eca8654e2f5d2d306b506f9d213cf1b